grails-app/services/com/unifina/service/DataUnionSecretService.groovy

```diff
@@ -22,9 +22,10 @@ class DataUnionSecretService {
 		return result
 	}
 
-	DataUnionSecret find(String contractAddress, String DataUnionSecretId) {
-		DataUnionSecret result = DataUnionSecret.where {
-			(contractAddress == contractAddress) && (id == DataUnionSecretId)
+	DataUnionSecret find(String contractAddress, String id) {
+		DataUnionSecret result = DataUnionSecret.createCriteria().get {
```

Comment:
What does .get().find() do?

Reply:
@hpihkala I was wondering the same but adding find() to problematic query doesn't resolve issue. I think it should be removed here.

Reply:
Yeah looks lie a leftover from old code. Elsewhere you're using just `.get()`.